### PR TITLE
Fix false positives for Less maps in property-no-unknown

### LINKED ISSUE #4834: Fix false positives for maps in property-no-unknown
> Clearly describe the bug

Rule **property-no-unknown** highlight syntax inside [LESS Maps](http://lesscss.org/features/#maps-feature). I don't understand why. 

[Screenshot with error](https://i.postimg.cc/J4zJNbP4/2020-06-19-10-04-49.png)

> Which rule, if any, is the bug related to?

`property-no-unknown `

> What code is needed to reproduce the bug?

```less
@sizes: {
  mobile: 320px; // -> Unexpected unknown property "mobile"
  tablet: 768px; // -> Unexpected unknown property "tablet"
  desktop: 1024px; // -> Unexpected unknown property "desktop"
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "property-no-unknown": "true"
  }
}
```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"Webstorm plugin"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

 "Yes, it's related to LESS maps."

> What did you expect to happen?

"No warnings inside LESS maps syntaxes."

> What actually happened (e.g. what warnings or errors did you get)?

 "The following warnings were flagged:"

```
Unexpected unknown property "..."
```

### PR DESCRIPTION
**Pull request by jrlmontejo**
> Which issue, if any, is this issue related to?

Closes #4834 

> Is there anything in the PR that needs further explanation?

None

### CODE REVIEW

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@jrlmontejo Thanks for the pull request, and apologies for just getting around to reviewing it.

Can you please:

- set to merge into the `v14` branch
- add a test to [`isStandardSyntaxDeclaration.test.js`](https://github.com/stylelint/stylelint/blob/v14/lib/utils/__tests__/isStandardSyntaxDeclaration.test.js)

I think we'll be good then.

**Comment by jrlmontejo** [comment]
@jeddy3 I applied the requested changes. 

Thanks!

**Review by jeddy3 — APPROVED** [review]
Thanks for making the changes. LGTM.

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

File: lib/utils/isStandardSyntaxDeclaration.js [hunk 1]
```diff
@@ -40,6 +40,11 @@ module.exports = function (decl) {
 		return false;
 	}
 
+	// Less map declaration
+	if (parent.type === 'atrule' && parent.raws.afterName === ':') {
```
Comment (ybiquitous) [inline lib/utils/isStandardSyntaxDeclaration.js:44]: I find type errors on the `v14` branch.

```console
$ npm --silent run lint:types
lib/utils/isStandardSyntaxDeclaration.js:45:6 - error TS2532: Object is possibly 'undefined'.

45  if (parent.type === 'atrule' && parent.raws.afterName === ':') {
        ~~~~~~

lib/utils/isStandardSyntaxDeclaration.js:45:34 - error TS2532: Object is possibly 'undefined'.

45  if (parent.type === 'atrule' && parent.raws.afterName === ':') {
                                    ~~~~~~


Found 2 errors.
```

How about adding the following check?

```suggestion
	if (parent && parent.type === 'atrule' && parent.raws.afterName === ':') {
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]

**Review by ybiquitous — APPROVED** [review]
@jrlmontejo Thank you! 👍🏼

**Comment by jeddy3** [comment]
v14 changelog entry:

> - Fixed: `property-no-unknown` false positives for Less maps ([#5381](https://github.com/stylelint/stylelint/pull/5381)).

**Pull request merged**